test/e2e/storage/volume_io.go

```diff
@@ -0,0 +1,822 @@
+// Note: `fsizes` values are expected to each be at least 1kb and a multiple of 1024 bytes.
+func testVolumeIO(f *framework.Framework, cs clientset.Interface, config framework.VolumeTestConfig, volsrc v1.VolumeSource, file string, fsizes []int) (err error) {
+	writeBlk := strings.Repeat("abcdefghijklmnopqrstuvwxyz123456", 32) // 1k file for `dd`
+	volName := fmt.Sprintf("%s-%s", config.Prefix, "io-volume")
+	dir := path.Join("/opt", config.Prefix, config.Namespace)
+	dd_input := path.Join(dir, "dd_if")
+	// setup init environment and keep pod alive until deleted
+	initCmd := fmt.Sprintf("mkdir -p %s; echo %s >%s; sleep 3600", dir, writeBlk, dd_input)
+
+	clientPod := &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: config.Prefix + "-io-client",
+			Labels: map[string]string{
+				"role": config.Prefix + "-io-client",
+			},
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:  config.Prefix + "-io-client",
+					Image: framework.BusyBoxImage,
+					Command: []string{
+						"/bin/sh",
+						"-c",
+						initCmd,
+					},
+					VolumeMounts: []v1.VolumeMount{
+						{
+							Name:      volName,
+							MountPath: dir,
+						},
+					},
+				},
+			},
+			SecurityContext: &v1.PodSecurityContext{
+				SELinuxOptions: &v1.SELinuxOptions{
+					Level: "s0:c0,c1",
+				},
+			},
+			Volumes: []v1.Volume{
+				{
+					Name:         volName,
+					VolumeSource: volsrc,
+				},
+			},
+		},
+	}
+
+	By(fmt.Sprint("starting ", config.Prefix, " io-client"))
+	podsNamespacer := cs.CoreV1().Pods(config.Namespace)
+	clientPod, err = podsNamespacer.Create(clientPod)
+	if err != nil {
+		return fmt.Errorf("Failed to create client pod %q: %v", clientPod.Name, err)
+	}
+	defer func() {
+		By(fmt.Sprintf("deleting client pod %q...", clientPod.Name))
+		delErr := framework.DeletePodWithWait(f, cs, clientPod)
+		if delErr != nil {
+			framework.Logf("client pod failed to delete: %v", delErr)
+			if err == nil { // delete err is returned if err is not set
+				err = delErr
+			}
+		}
+		// hack so kubelet has time to delete vol mounts before we delete the server pod
+		time.Sleep(20 * time.Second)
+	}()
+
+	err = framework.WaitForPodRunningInNamespace(cs, clientPod)
+	if err != nil {
+		return fmt.Errorf("client pod %q not running: %v", clientPod.Name, err)
+	}
+
+	// create files of the passed-in file sizes
+	for _, fsize := range fsizes {
+		// file sizes must be a multiple of the length of the `writeBlk` value (1024 currently)
+		if fsize < len(writeBlk) {
+			fsize = len(writeBlk)
+		} else if fsize/len(writeBlk)*len(writeBlk) != fsize {
+			fsize = fsize/len(writeBlk) + len(writeBlk)
+		}
+		fpath := path.Join(dir, fmt.Sprintf("%s-%d", file, fsize))
+
+		By(fmt.Sprintf("Writing %d bytes to test file %s", fsize, fpath))
+		// dd's bs*seek = size (note this creates the file 1 byte more than the target size)
+		seek := fsize/len(writeBlk) - 1
```

Comment:
Yeah, I meant instead of generating the content in the e2e test program, have the test program invoke some script that will generate the content.  That avoids sending the content in podExec.  I would try that and then see if you are able to generate enough io that way.  I think the main issue is that you need to use a big enough file so that the overhead of reading the file into memory is small compared to writing it back out.

If that doesn't work, another way to exercise the i/o would be to write a simple go program that just continuously writes.  But then you would need to find a place to put the container image.